src/sentry/migrations/0287_backfill_snubaquery_environment.py

```diff
@@ -0,0 +1,113 @@
+# Generated by Django 2.2.27 on 2022-04-06 22:58
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_snubaquery_environment(apps, schema_editor):
+    # the current snuba_query.environment_id could be referencing an environment it shouldn't be:
+    # during project transfer, the AlertRule changed organization_id's but some work wasn't done:
+    # 1. Conditionally create the environment from the old org to the new org
+    # 2. Update the environment_id for the SnubaQuery to reference the new org environment
+    # So now we have SnubaQueries that reference(fk to Environment) environments from the old org.
+
+    SnubaQuery = apps.get_model("sentry", "SnubaQuery")
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    Project = apps.get_model("sentry", "Project")
+    AlertRule = apps.get_model("sentry", "AlertRule")
+
+    for snuba_query in RangeQuerySetWrapperWithProgressBar(
+        SnubaQuery.objects.all().select_related("environment")
+    ):
+        if not snuba_query.environment:
+            continue
+
+        snuba_env = snuba_query.environment
+
+        try:
+            alert_rule = AlertRule.objects_with_snapshots.filter(
+                snuba_query_id=snuba_query.id
+            ).get()
+            if alert_rule.organization_id == snuba_env.organization_id:
+                continue
+        except AlertRule.DoesNotExist:
+            continue
+
+        envs_by_org = Environment.objects.filter(
+            organization_id=alert_rule.organization_id, name=snuba_env.name
+        ).distinct()
```

Comment:
Didn't think of using the alert rule's org id, good call

Reply:
Lol, yeah this was the bit that was causing trouble for my test ...